### ISSUE TITLE
Is shikigva=40 still supported

### ISSUE THREAD

**Issue by superkid200** [dortania/bugtracker#96]
Is shikigva=40 supported in macOS Big Sur? Because this swaps board-id with iMacPro14,2 (which is no longer supported starting with macOS Big Sur).

I have an AMD CPU, so I use iMacPro1,1 just like anybody else (however that cause crashes in the setup process: IOGraphicsFamily.kext causes kernel panics).

Link: [GPU Buyers Guide - NVIDIA boot args](https://dortania.github.io/GPU-Buyers-Guide/misc/bootflag.html#nvidia-boot-arguments)

Possible fixes: Test it then add information about whether it works or not.

**Comment by khronokernel** [comment]
Well the issue is shikigva is a userspace patch which Lilu currently does not support in Big Sur. Assuming when it's fixed, shikigva=40 will still be a valid flag however this is going to be dependant on how other kexts are coded. Currently AppleGraphicsDevicePolicy still has a profile setup for Mac-27ADBB7B4CEE8E61(iMac14,2) but it's quite likely code like this will be removed in the coming months.

I believe internally we were deciding to recommend MacPro7,1 For AMD users with Nvidia GPUs, however we haven't updated the PlatformInfo pages. With the shiki flag, we'll wait for Lilu's userspace patcher to be fixed and go from there

**Comment by superkid200** [comment]
I have seen some people have been able to use shikigva boot flag (even though it is for AMD GPU). So wonder how shikigva=40 does not work.

**Comment by khronokernel** [comment]
shikigva=40 does multiple things, looking closely at [kern_shiki.hpp](https://github.com/acidanthera/WhateverGreen/blob/master/WhateverGreen/kern_shiki.hpp):

```cpp
// Unlike 10.12.6 without security updates and earlier, on 10.13 and latest 10.12.6 AppleGVA patches
// do not apply to all processes, and each process needs to be patched explicitly. This is a bug
// in Lilu, which needs to be explored and fixed. For now this bit ensures that the processes present
// in WHITELIST section of Patches.plist will definitely get the fixes even on 10.13 and 10.12.6.
// On 10.12.6 and 10.13 this must be used if any of the following bits are used:
// - ForceOnlineRenderer
// - AllowNonBGRA
// - ForceCompatibleRenderer
// - ReplaceBoardID
// - FixSandyBridgeClassName
// It is enabled automatically on 10.12 and 10.13 if shikigva is *NOT* passed and ForceCompatibleRenderer or
// FixSandyBridgeClassName are automatically enabled.
AddExecutableWhitelist     = 8,

// Replace board-id used by AppleGVA and AppleVPA by a different board-id.
// Sometimes it is feasible to use different GPU acceleration settings from the main mac model.
// By default Mac-27ADBB7B4CEE8E61 (iMac14,2) will be used, but you can override this via shiki-id boot-arg.
// See /System/Library/PrivateFrameworks/AppleGVA.framework/Resources/Info.plist for more details.
ReplaceBoardID             = 32,
```

We can see the patch would be both user and kernelspace based, however odd that it's working "partially". I would've assumed all functionality would be disabled, however I haven't looked into the rest of Shiki's source in some time
